src/swarm-engine/Swarm/Game/State.hs

```diff
@@ -161,8 +160,9 @@ parseCodeFile ::
   FilePath ->
   m CodeToRun
 parseCodeFile filepath = do
-  contents <- sendIO $ TIO.readFile filepath
-  pt <- either (throwError . CustomFailure) return (processTermEither contents)
+  mcontents <- sendIO $ readFileMayT SystemLocale filepath
+  contents <- maybe (throwError (AssetNotLoaded (Data Script) filepath (DoesNotExist File))) pure mcontents
```

Comment:
Is there a function to run an action and handle the alternative? 🤔 

I imagine something like:
```Haskell
contents <- sendIO (readFileMayT SystemLocale filepath)
  ??? throwError (AssetNotLoaded (Data Script) filepath (DoesNotExist File))
```

Reply:
There's no standard one that I know of, though apparently it's been redefined a bunch of times in various projects:

https://hoogle.haskell.org/?hoogle=m%20(Maybe%20a)%20-%3E%20m%20a%20-%3E%20m%20a